submariner/globalnet-enhancement2-0.md

```diff
@@ -0,0 +1,282 @@
+  annotated with a globalIP.
+* For Headless Services, necessary ingress rules will be programmed to support connectivity to
+  the backend Pods. In a Headless Service use-case, every backend Pod will be annotated with
+  a unique globalIP as an Ingress IP. Clients can discover the associated globalIP using
+  Lighthouse and can connect to the globalIP assigned to the backend pods.
+  For Pods backed by Headless Services, if the user does not explicitly request a globalIP via
+  GlobalnetEgressIP CRD, the same globalIP assigned to the Pods will be used for both Ingress
+  as-well-as Egress. However, if user creates a GlobalnetEgressIP that matches the namespace,
+  or the backend pods of Headless Service, the globalIP assigned via GlobalnetEgressIP takes
+  precedence over individual Pod globalIPs as EgressIP.
+
+![GlobalnetTopology](./images/globalnet-topology.png)
+
+If the external application in a remote Cluster is interested in consuming the exported
+Service, it will have to reach the globalIP assigned to the exportedService and the response
+from the backend Pods will have the source as the globalIP of the exportedService.
+
+On the other hand, when the backend Pod tries to talk to the external application in remote
+Cluster, the external application will see that traffic is coming with a globalIP that is
+assigned either at the Cluster level, or namespace level, or the one requested by the user
+at Pod level. This asynchronous behavior will be analogous to the behavior seen with standard
+K8s applications when they talk to any Services. Applications desiring to have the same IPaddress
+for both ingress and egress traffic can create Headless Services in front of them and should
+avoid requesting explicit globalIP that match Headless Service backend Pods.
+
+The proposal has its own set of Pros and Cons.
+
+## Pros
+
+Since we will be annotating only select K8s objects with globalIP, it will help in improved scalability.
+Also, this translates to fewer iptable rules on the active Gateway node of the cluster which will
+improve the overall performance of the solution.
+
+In the current implementation, when a client Pod is scheduled, until it's annotated with a globalIP,
+and the corresponding egress rules are programmed on the node, it was not able to communicate with
+remote Services. This was creating some issues in-terms of user-experience. In this proposal, as we
+have a globalIP at the cluster level, a Pod can communicate with remote Services right away which
+will be closer to the Vanilla Submariner experience.
+
+## Cons
+
+A limitation with the proposed approach is that we will lose the unique identity for **every**
+Pod which is supported today. For users that intend to use Network Policies across Clusters or
+for debugging purposes, this can be mitigated by creating the necessary GlobalnetEgressIP CRDs
+which create unique globalIPs at the Namespace/Pod level.
+
+The following CRD is proposed to support the new design.
```

Comment:
Shouldn't this be in a section?
Right now it'll be under "Cons"

Reply:
Done